networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRangeMessageHandler.java

```diff
@@ -183,7 +187,7 @@ private SafeFuture<RequestState> sendMatchingBlocks(
                       .map(MinimalBeaconBlockSummary::getSlot)
                       .orElse(ZERO);
               final NavigableMap<UInt64, Bytes32> hotRoots;
-              if (combinedChainDataClient.isFinalized(endSlot)) {
+              if (combinedChainDataClient.isFinalized(message.getMaxSlot())) {
```

Comment:
I'm not quite sure this will end up in the same number as the previous approach used in line 168.

Reply:
The `getMaxSlot` gives a different result but is correct. IMU the `endSlot` calculation wasn't correct and returned `maxSlot + 1`. The first blocks (from the requested count) is always at the `startSlot` so we should first subtract 1 from the `count` to calculate the `endSlot`/`maxSlot`